src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const history = [];
+  let newState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        newState = { ...newState, ...action.extraData };
+        break;
+      case 'removeProperties':
+        newState = { ...newState };
```

Comment:
This is an unnecessary string, it must be removed. That doesn't have any sense, you just make a copy of this object in this object.